### DISCUSSION TITLE
ClientScope - Keycloak API Docu vs. Admin-client vs. Admin-ui

### DISCUSSION THREAD

**Discussion by AmusedShadow** [keycloak/keycloak#16086] [Q&A]
In case of creating with the admin ui a new client scope, it get's an type.
```
{
    "name": "test",
    "description": "",
    "attributes": {
        "consent.screen.text": "",
        "display.on.consent.screen": "true",
        "include.in.token.scope": "true",
        "gui.order": ""
    },
    "type": "optional",
    "protocol": "openid-connect"
}
```

But this property type is neither part of the ClientScopeRepresentation within the admin-client package nor it is part of the official REST API Docu ClientScopeRepresentation ... 
https://www.keycloak.org/docs-api/14.0/rest-api/#_clientscoperepresentation

Is it now a bug, a hidden feature or just some whit noise ?

**Comment by edewit** [comment]
in the UI it's shown as a property of scope, but it's not. There are separate endpoints to control the `type`: namely `/default-default-client-scopes` and `/default-optional-client-scopes`

**Comment by AmusedShadow** [comment]
Ah ok. Because the ui sends within the create request. That was a little confusing :)
